### PR TITLE
add ccc-08-s4 solution

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/ccc-Twenty-four.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		}
+	}
+	return INT_MIN;
+}
+
+int main() {
+	int num_hands;
+	cin >> num_hands;
+
+	while (num_hands--) {
```
Comment (SansPapyrus683) [inline solutions/ccc-Twenty-four.mdx:46]:
```suggestion
	for (int h = 0; h < num_hands; h++) {
```

File: solutions/ccc-Twenty-four.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			cin >> hand[i];
+		}
+		sort(hand.begin(), hand.end());
+
+		int ans = INT_MIN;
+		/*
+		 * We go though all possible permutations of the current hand and check
+		 * the result of the expression for each possible way of arranging the
+		 * arithmetic operations.
+		 */
```
Comment (SansPapyrus683) [inline solutions/ccc-Twenty-four.mdx:58]: for long comments like this, it's better to just put them in the editorial and give a brief description here

File: solutions/ccc-Twenty-four.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		do {
+			/*
+			 * There are two possible ways that parentheses can be arranged:
+			 * (((1 + 1) + 1) + 1), or ((1 + 1) + (1 + 1)). We all possible
```
Comment (SansPapyrus683) [inline solutions/ccc-Twenty-four.mdx:62]: same with this one

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/ccc-Twenty-four.mdx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+// Function that takes in two numbers and an operation and returns the result.
+int operation(int op, int num1, int num2) {
+	switch (op) {
+		case 0:
+			return num1 + num2;
+		case 1:
+			return num1 - num2;
+		case 2:
+			return num1 * num2;
+		case 3: {
```
Comment (Aadit-Ambadkar) [inline solutions/ccc-Twenty-four.mdx:34]: Dam I haven't seen these in forever (switch case)

File: solutions/ccc-Twenty-four.mdx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+						}
+					}
+				}
+			}
+
+			// Case 2: (( ) ( ))
+			for (int i = 0; i < 4; i++) {
+				for (int j = 0; j < 4; j++) {
+					for (int k = 0; k < 4; k++) {
+						int first = operation(i, hand[0], hand[1]);
```
Comment (Aadit-Ambadkar) [inline solutions/ccc-Twenty-four.mdx:90]: I think this is 92 characters...

File: solutions/ccc-Twenty-four.mdx [hunk 6]
```diff
@@ -7,7 +7,10 @@ author: Paul Chen
 
 ## Explanation
 
-We can complete search through all possible ways to create an arithmetic expression with a card hand. For each permutation of the card hand, we try every way of placing arithmetic operators. The arithmetic expression with the maximum value that is not greater than 24 is our answer.
+We can complete search through all possible ways to create an arithmetic expression with a card hand. 
+For each permutation of the card hand, we try every way of placing arithmetic operators. 
+Note that there are two possible ways that parentheses can be arranged: ((( ) ) ), or (( ) ( )), so we must try both cases.
```
Comment (SansPapyrus683) [inline solutions/ccc-Twenty-four.mdx:12]: put the parenthesis parts in single line code?

File: solutions/ccc-Twenty-four.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/ccc-Twenty-four.mdx:34]: same

File: solutions/ccc-Twenty-four.mdx (hunk 6 above)
Comment (876pol) [reply solutions/ccc-Twenty-four.mdx:12]: oh good idea

File: solutions/ccc-Twenty-four.mdx (hunk 5 above)
Comment (876pol) [reply solutions/ccc-Twenty-four.mdx:90]: i think usaco guide shows each tab as four spaces, so it should still fit

File: solutions/ccc-Twenty-four.mdx (hunk 4 above)
Comment (876pol) [reply solutions/ccc-Twenty-four.mdx:34]: switch case > if else

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/ccc-Twenty-four.mdx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+		case 0:
+			return num1 + num2;
+		case 1:
+			return num1 - num2;
+		case 2:
+			return num1 * num2;
+		case 3: {
+			// The divisor cannot be 0 and the quotient must be a whole number.
+			if (num2 == 0 || num1 % num2) {
+				return INT_MIN;
```
Comment (Aadit-Ambadkar) [inline solutions/ccc-Twenty-four.mdx:37]: no, don't use INT_MIN. Create a small constant (-1e9 or smth) and use that instead. INT_MIN is prone to overflow errors, and is bad practice.

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]
so many for loops I wish c++ had a way to simply it.

File: solutions/ccc-Twenty-four.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply solutions/ccc-Twenty-four.mdx:37]: no??? if so give an SO link
if you're doing things right this shouldn't overflow

Comment (876pol) [reply solutions/ccc-Twenty-four.mdx:37]: oh ok well I'll leave it as-is for now

**Pull request closed without merging**